OSOE-53: RouteModel, unit test fixes, orchard-core-preview (#95)
Lombiq/Helpful-Libraries: Lombiq.HelpfulLibraries/Docs/MvcLibraries.md

- `UrlHelperExtensions`: Adds extension methods to the `@Url` helper, such as `EditContentItemWithTab`.
- `ShapeResultExtensions`: Adds extensions methods generating placement strings on your shape description, such as `UseTab`.

## `RouteModel`

> This class provides a strongly typed way to generate local URLs for Orchard Core MVC actions. It uses lambda expressions to select the action and provide arguments. Use `RouteModel.CreateFromExpression<TClass>(...).ToString()` or the provided `OrchardHelper.Action` and `HttpContext.Action` extensions. If you also use our [UI Testing Toolbox](https://github.com/Lombiq/UI-Testing-Toolbox/) that has a `UITestContext.GoTo` extension method as well.   

Reviewer: ```suggestion
This class provides a strongly typed way to generate local URLs for Orchard Core MVC actions. It uses lambda expressions to select the action and provide arguments. Use `RouteModel.CreateFromExpression<TClass>(...).ToString()` or the provided `OrchardHelper.Action()` and `HttpContext.Action()` extensions. If you also use our [UI Testing Toolbox](https://github.com/Lombiq/UI-Testing-Toolbox/) that has a `UITestContext.GoTo()` extension method as well.   

```

> This class provides a strongly typed way to generate local URLs for Orchard Core MVC actions. It uses lambda expressions to select the action and provide arguments. Use `RouteModel.CreateFromExpression<TClass>(...).ToString()` or the provided `OrchardHelper.Action` and `HttpContext.Action` extensions. If you also use our [UI Testing Toolbox](https://github.com/Lombiq/UI-Testing-Toolbox/) that has a `UITestContext.GoTo` extension method as well.   

Reviewer 2: ```suggestion
This class provides a strongly typed way to generate local URLs for Orchard Core MVC actions. It uses lambda expressions to select the action and provide arguments. Use `RouteModel.CreateFromExpression<TClass>(...).ToString()` or the provided `OrchardHelper.Action()` and `HttpContext.Action()` extensions. If you also use our [UI Testing Toolbox](https://github.com/Lombiq/UI-Testing-Toolbox/), you can use the `UITestContext.GoTo()` extension method from there, as well. 
```
